web/modules/custom/bnf/bnf_client/src/Form/BnfSubscriptionCreateForm.php

```diff
@@ -0,0 +1,239 @@
+    $form['tags'] = $this->getTermFormElement('tags');
+    $form['tags']['#disabled'] = !$isEditable;
+
+    $form['actions']['submit'] = [
+      '#type' => 'submit',
+      '#disabled' => !$isEditable,
+      '#value' => $this->t('Create subscription', [], ['context' => 'BNF']),
+    ];
+
+    return $form;
+  }
+
+  /**
+   * Building a term-reference form element.
+   *
+   * @return mixed[]
+   *   A form element, to be used in $form.
+   */
+  private function getTermFormElement(string $vid): array {
+    $options = [];
+
+    $terms = $this->termStorage->loadByProperties(['vid' => $vid]);
+
+    foreach ($terms as $term) {
+      $options[$term->id()] = $term->label();
+    }
+
+    return [
+      '#title' => $this->t('Map content to @vocabulary', ['@vocabulary' => $vid], ['context' => 'BNF']),
+      '#description' => $this->t('When content is added through this subscription, these terms will be automatically added. <strong>If the content only supports a single term, only the first will be added.</strong>', [], ['context' => 'BNF']),
+      "#type" => "select2",
+      "#options" => $options,
+      "#multiple" => TRUE,
+      "#target_type" => "taxonomy_term",
+      "#selection_handler" => "default:taxonomy_term",
+      "#selection_settings" => [
+        "match_operator" => "CONTAINS",
+        "match_limit" => 10,
+        "sort" => [
+          "field" => "name",
+        ],
+        "target_bundles" => [
+          $vid => $vid,
+        ],
+      ],
+    ];
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  public function validateForm(array &$form, FormStateInterface $form_state): void {
+    $uuid = $form_state->get('uuid');
+    $existingSubscriptions = $this->subscriptionStorage->loadByProperties(['subscription_uuid' => $uuid]);
+
+    if (!empty($existingSubscriptions)) {
+      $form_state->setErrorByName(
+        'uuid',
+        $this->t('Could not create subscription - this subscription already exists.', [], ['context' => 'BNF'])
+      );
+    }
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  public function submitForm(array &$form, FormStateInterface $form_state): void {
+    $uuid = $form_state->get('uuid');
+    $existingSubscriptions = $this->subscriptionStorage->loadByProperties([
+      'subscription_uuid' => $uuid,
+    ]);
+
+    // Even though we already have this check as part of the form building,
+    // we want to make sure that even if the form submission is triggered anyway
+    // we don't end up with duplicate subscriptions.
+    if (!empty($existingSubscriptions)) {
+      $form_state->setErrorByName('uuid', $this->t('This subscription already exists.', [], ['context' => 'BNF']));
+      return;
+    }
+
+    try {
+      /** @var \Drupal\bnf_client\Entity\Subscription $subscription */
+      $subscription = $this->subscriptionStorage->create([
+        'subscription_uuid' => $uuid,
+      ]);
+
+      if ($form_state->getValue('only_new_content')) {
+        $subscription->setLast(time());
```

Comment:
`time()` is frowned upon. See `Drupal\Component\Datetime\Time`, the `datetime.time` service.